python/ml4ir/base/model/optimizer.py

```diff
@@ -37,38 +24,83 @@ def get_optimizer(
-        return tf_optimizers.RMSprop(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
+    if 'lr_schedule' not in model_config:
+        learning_rate_schedule = ExponentialDecay(
+            initial_learning_rate=0.01,
+            decay_steps=10000000,
+            decay_rate=1.0,
+            staircase=True,
         )
+
+    else:
+        lr_schedule = model_config['lr_schedule']
+        lr_schedule_key = lr_schedule['key']
+        if lr_schedule_key == LearningRateScheduleKey.EXPONENTIAL:
+            learning_rate_schedule = ExponentialDecay(
+                initial_learning_rate=lr_schedule['learning_rate'] if 'learning_rate' in lr_schedule else 0.01,
+                decay_steps=lr_schedule['learning_rate_decay_steps'] if 'learning_rate_decay_steps' in lr_schedule else 10000000,
+                decay_rate=lr_schedule['learning_rate_decay'] if 'learning_rate_decay' in lr_schedule else 1.0,
+                staircase=True,
+            )
+
+        elif lr_schedule_key == LearningRateScheduleKey.CYCLIC:
+            lr_schedule_type = lr_schedule['type']
+            if lr_schedule_type == CyclicLearningRateType.TRIANGULAR:
+                learning_rate_schedule = cyclic_learning_rate.TriangularCyclicalLearningRate(
+                    initial_learning_rate=lr_schedule['initial_learning_rate'] if 'initial_learning_rate' in lr_schedule else 0.001,
+                    maximal_learning_rate=lr_schedule['maximal_learning_rate'] if 'maximal_learning_rate' in lr_schedule else 0.01,
+                    step_size=lr_schedule['step_size'] if 'step_size' in lr_schedule else 10,
+                )
+            elif lr_schedule_type == CyclicLearningRateType.TRIANGULAR2:
+                learning_rate_schedule = cyclic_learning_rate.Triangular2CyclicalLearningRate(
+                    initial_learning_rate=lr_schedule['initial_learning_rate'] if 'initial_learning_rate' in lr_schedule else 0.001,
+                    maximal_learning_rate=lr_schedule['maximal_learning_rate'] if 'maximal_learning_rate' in lr_schedule else 0.01,
+                    step_size=lr_schedule['step_size'] if 'step_size' in lr_schedule else 10,
+                )
+            elif lr_schedule_type == CyclicLearningRateType.EXPONENTIAL:
+                learning_rate_schedule = cyclic_learning_rate.ExponentialCyclicalLearningRate(
+                    initial_learning_rate=lr_schedule['initial_learning_rate'] if 'initial_learning_rate' in lr_schedule else 0.001,
+                    maximal_learning_rate=lr_schedule['maximal_learning_rate'] if 'maximal_learning_rate' in lr_schedule else 0.01,
+                    step_size=lr_schedule['step_size'] if 'step_size' in lr_schedule else 10,
+                    gamma=lr_schedule['gamma'] if 'gamma' in lr_schedule else 1.0,
+                )
+            else:
+                raise ValueError("illegal cyclic learning rate schedule type key: " + lr_schedule_type)
```

Comment:
Let's start the ValueError with a capital. Also, we can make it even more informative e.g.,: 
`raise ValueError("Unsupported learning rate schedule type key: {}. 
We currently support between [{}].".format(lr_schedule_type, " ".join(CyclicLearningRateType.get_all_calues()))`